docs(linter): Fix `jsx-a11y/img-redundant-alt` configuration option names.

connorshea opened the pull request:
These were wrong previously due to the fields having different names in the config object than the actual configuration passed to the rule. The docs will now match the result we expect to see from the 

Also updated the tests to cover more cases, and found a few places where the current logic is problematic.

Generated docs:

```md
## Configuration

This rule accepts a configuration object with the following properties:

### components

type: `string[]`

default: `["img"]`

JSX element types to validate (component names) where the rule applies.
For example, `["img", "Image"]`.

### words

type: `string[]`

default: `["image", "photo", "picture"]`

Words considered redundant in alt text that should trigger a warning.
```

camc314 commented:
### Merge activity

* **Dec 7, 2:41 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 7, 2:41 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Fix `jsx-a11y/img-redundant-alt` configuration option names. (#16552)

These were wrong previously due to the fields having different names in the config object than the actual configuration passed to the rule. The docs will now match the result we expect to see from the

Also updated the tests to cover more cases, and found a few places where the current logic is problematic.

Generated docs:

```md
## Configuration

This rule accepts a configuration object with the following properties:

### components

type: `string[]`

default: `["img"]`

JSX element types to validate (component names) where the rule applies.
For example, `["img", "Image"]`.

### words

type: `string[]`

default: `["image", "photo", "picture"]`

Words considered redundant in alt text that should trigger a warning.
```
